File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
       int left = 0;
       int right = height.length-1;
       int area = 0;
       while(left < right){
        int width = right - left;
        int length = Math.min(height[left],height[right]);
        int temparea = length*width;
        if(temparea > area)
            area = temparea;
        if(height[left] < height[right])
            left++;
        else
            right--;
        }
    return area;
    }

}

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int low = 0,high = height.size()-1;

        int maxArea = 0;

        while(low < high){

            int curArea = min(height[low],height[high]) * (high - low);

            maxArea = max(maxArea,curArea);

            if(height[low] < height[high]) low++;
            else high--;

        }

        return maxArea;
    }
};